src/textual/screen.py

```diff
@@ -479,20 +475,12 @@ def _on_timer_update(self) -> None:
                 self._dirty_widgets.clear()
 
         if self._callbacks:
-            self.post_message(events.InvokeCallbacks())
-
-    async def _on_invoke_callbacks(self, event: events.InvokeCallbacks) -> None:
-        """Handle PostScreenUpdate events, which are sent after the screen is updated"""
-        await self._invoke_and_clear_callbacks()
+            self.call_next(self._invoke_and_clear_callbacks)
 
     async def _invoke_and_clear_callbacks(self) -> None:
         """If there are scheduled callbacks to run, call them and clear
         the callback queue."""
         if self._callbacks:
-            display_update = self._compositor.render_update(
```

Comment:
Suspect this was a bit of a hail Mary to ensure that the callbacks happened after a refresh. I'm 99% certain this isn't required now.